### ISSUE TITLE
您好，windows使用TensorRT-8.2.1.8，部署paddleocr表格识别模型ch_ppstructure_mobile_v2.0_SLANet_infer.onnx，初始化报错

### ISSUE THREAD

**Issue by cdycdycdy** [NVIDIA/TensorRT#2760]
[03/13/2023-14:48:36] [W] [TRT] TensorRT was linked against cuBLAS/cuBLAS LT 11.6.3 but loaded cuBLAS/cuBLAS LT 11.1.0
[03/13/2023-14:48:36] [W] [TRT] TensorRT was linked against cuDNN 8.2.1 but loaded cuDNN 8.1.0
[03/13/2023-14:49:05] [W] [TRT] Myelin graph with multiple dynamic values may have poor performance if they differ. Dynamic values are:
[03/13/2023-14:49:05] [W] [TRT]  (# 3 (SHAPE x))
[03/13/2023-14:49:05] [W] [TRT]  (# 0 (SHAPE x))
[03/13/2023-14:49:08] [W] [TRT] Skipping tactic 0 due to insuficient memory on requested size of 1514143744 detected for tactic 0.
[03/13/2023-14:49:08] [E] [TRT] 10: [optimizer.cpp::nvinfer1::builder::`anonymous-namespace'::LeafCNode::computeCosts::2011] Error Code 10: Internal Error (Could not find any implementation for node {ForeignNode[linear_13.w_0 + (Unnamed Layer* 321) [Shuffle]...p2o.Reshape.23 + p2o.Transpose.7]}.)
[ch_ppstructure_mobile_v2.0_SLANet_infer.onnx.zip](https://github.com/NVIDIA/TensorRT/files/10954419/ch_ppstructure_mobile_v2.0_SLANet_infer.onnx.zip)

**Comment by zerollzeng** [comment]
does it work with onnxruntime? Thanks!

**Comment by cdycdycdy** [comment]
> what's the input shape?

{ 1, 3, 488, 488 }

**Comment by cdycdycdy** [comment]
> does it work with onnxruntime? Thanks!

yes, it's right in onnxruntime

**Comment by zerollzeng** [comment]
@nvpohanh @zhenhuaw-me test with latest 8.6
```
[03/16/2023-07:26:12] [E] Error[2]: [myelinBuilderUtils.cpp::getMyelinSupportType::1266] Error Code 2: Internal Error (ForeignNode does not support data-dependent shape for now.)
[03/16/2023-07:26:12] [E] Engine could not be created from network
[03/16/2023-07:26:12] [E] Building engine failed
[03/16/2023-07:26:12] [E] Failed to create engine from model or file.
[03/16/2023-07:26:12] [E] Engine set up failed
&&&& FAILED TensorRT.trtexec [TensorRT v8600] # trtexec --onnx=ch_ppstructure_mobile_v2.0_SLANet_infer.onnx --optShapes=x:1x3x488x488 --verbose
```

Is it a known issue? do you want me to create an internal bug?

**Comment by cdycdycdy** [comment]
> @nvpohanh @zhenhuaw-me test with latest 8.6
> 
> ```
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.68 with p2o.BatchNormalization.64
> [03/16/2023-07:26:12] [V] [TRT] Running: ConvScaleFusion on p2o.Conv.69
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.69 with p2o.BatchNormalization.65
> [03/16/2023-07:26:12] [V] [TRT] Running: ConvScaleFusion on p2o.Conv.70
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.70 with p2o.BatchNormalization.66
> [03/16/2023-07:26:12] [V] [TRT] Running: ConvScaleFusion on p2o.Conv.71
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.71 with p2o.BatchNormalization.67
> [03/16/2023-07:26:12] [V] [TRT] Running: ConvScaleFusion on p2o.Conv.72
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.72 with p2o.BatchNormalization.68
> [03/16/2023-07:26:12] [V] [TRT] Running: ConvScaleFusion on p2o.Conv.73
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.73 with p2o.BatchNormalization.69
> [03/16/2023-07:26:12] [V] [TRT] Running: ConvScaleFusion on p2o.Conv.74
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.74 with p2o.BatchNormalization.70
> [03/16/2023-07:26:12] [V] [TRT] Running: ConvScaleFusion on p2o.Conv.75
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.75 with p2o.BatchNormalization.71
> [03/16/2023-07:26:12] [V] [TRT] Running: ConvScaleFusion on p2o.Conv.76
> [03/16/2023-07:26:12] [V] [TRT] ConvScaleFusion: Fusing p2o.Conv.76 with p2o.BatchNormalization.72
> [03/16/2023-07:26:12] [V] [TRT] Running: ShuffleShuffleFusion on p2o.Reshape.8
> [03/16/2023-07:26:12] [V] [TRT] ShuffleShuffleFusion: Fusing p2o.Reshape.8 with p2o.Transpose.0
> [03/16/2023-07:26:12] [V] [TRT] Running: ShuffleShuffleFusion on (Unnamed Layer* 896) [Shuffle]
> [03/16/2023-07:26:12] [V] [TRT] ShuffleShuffleFusion: Fusing (Unnamed Layer* 896) [Shuffle] with p2o.Transpose.1
> [03/16/2023-07:26:12] [V] [TRT] Running: ShuffleErasure on (Unnamed Layer* 1223) [Shuffle]
> [03/16/2023-07:26:12] [V] [TRT] Removing (Unnamed Layer* 1223) [Shuffle]
> [03/16/2023-07:26:12] [E] Error[2]: [myelinBuilderUtils.cpp::getMyelinSupportType::1266] Error Code 2: Internal Error (ForeignNode does not support data-dependent shape for now.)
> [03/16/2023-07:26:12] [E] Engine could not be created from network
> [03/16/2023-07:26:12] [E] Building engine failed
> [03/16/2023-07:26:12] [E] Failed to create engine from model or file.
> [03/16/2023-07:26:12] [E] Engine set up failed
> &&&& FAILED TensorRT.trtexec [TensorRT v8600] # trtexec --onnx=ch_ppstructure_mobile_v2.0_SLANet_infer.onnx --optShapes=x:1x3x488x488 --verbose
> ```
> 
> Is it a known issue? do you want me to create an internal bug?

yes,i want to know  if this problem can be solved,and how to solved

**Comment by nvpohanh** [comment]
> Skipping tactic 0 due to insuficient memory on requested size of 1514143744 detected for tactic 0.

This means the workspace is not large enough. Try adding `--workspace=2048`

**Comment by cdycdycdy** [comment]
> 
adding --workspace=2048,but it cant's solve the problem

**Comment by nvpohanh** [comment]
> Error[2]: [myelinBuilderUtils.cpp::getMyelinSupportType::1266] Error Code 2: Internal Error (ForeignNode does not support data-dependent shape for now.)

Ah I see this error now. It seems that this model has some ops whose output shapes depend on input values. We currently have limited support for data-dependent shapes.

**Comment by nvpohanh** [comment]
It seems that the issue is the `OneHot` op inside the Loop. This is currently not supported. cc @rajeevsrao

**Comment by cdycdycdy** [comment]
> It seems that the issue is the `OneHot` op inside the Loop. This is currently not supported. cc @rajeevsrao

ok.Thanks very much
